Bootconfig fixes/improvements

Natolumin opened the pull request:
PR #65 refactored the bootconfig implementation, and in the process made it so we only load the kexec kernel, but don't boot it. This will be bundled with the fix for #77 since this code is unreachable because of that at the moment anyway

Natolumin committed:
bootconfig: Reboot after loading kernel

PR #65 refactored the bootconfig implementation, and in the process made
it so we only load the kexec kernel, but don't boot it

Natolumin committed:
bootconfig: Make Boot() fail on reboot failure

If we return from Reboot() without an error, there is something wrong as
it should cause the machine to reboot and thus never return. This logs a
failure and returns an error in such a case

insomniacslk commented:
Thanks @Natolumin . WIP because testing, or because you plan to make more changes?

insomniacslk commented:
@Natolumin I've updated the branch via UI, you may need to `git pull` before making further changes

Natolumin commented:
both
I intend to add whatever changes we decide on for the kexecbin issue; I'll add a log of the error condition (this function should never return with `err==nil` since we're supposed to reboot if all goes well)

and separately I'm running into weird bugs with the booted kernel panicking that might be caused by us passing incorrect arguments

Natolumin committed:
bootconfig: Catch more cases of missing KexecBin

When kexecbin is nowhere in PATH, the error we obtain from exec is
ErrNotFound, rather than something caught by os.IsNotExist(). Add a
second case to catch those

Natolumin commented:
Now I get the right error on boot, and it boots fine:
```
2018/12/04 18:16:33 BootConfig: KexecBin is not available, trying pure-Go kexec. Error: exec: "kexecbin": executable file not found in $PATH
```

insomniacslk commented:
From the title change I assume it's ready to review. @Natolumin do you confirm?

insomniacslk reviewed (approved):
Looks good to me, thanks!
